libraries/annotationprocessorconsumerkotlin/build.gradle

```
@@ -0,0 +1,10 @@
+apply plugin: "kotlin"
+
```

```suggestion
apply plugin: "kotlin-kapt"
```

you need to use `kapt` for kotlin libraries. annotationProcessor doesn't work.